### ISSUE TITLE
BSOD

### ISSUE THREAD

**Issue by shainodev** [UnamSanctam/SilentCryptoMiner#1076]
The bsod does not work when i try using lolminers thing (i dont know if it is lolminers fault).
![image](https://user-images.githubusercontent.com/77803435/186780496-f5c83086-1af1-4d84-a002-f1e6c3146f09.png)
this is the code idk whats wrong (i put the thread.sleep inside the function)

**Comment by shainodev** [comment]
it gets the pid
![image](https://user-images.githubusercontent.com/77803435/186780877-1eb20780-37b5-48d5-b7c0-6569c3173398.png)
but when i terminate it with task manager or process hacker nothing happends

**Comment by UnamSanctam** [comment]
Are you sure that you're running it as administrator? If you are then you can try another process to see if it's because something lolminer is doing.

**Comment by varysuzuki** [comment]
> Are you sure that you're running it as administrator? If you are then you can try another process to see if it's because something lolminer is doing.

OHHHHH admin is needed i didnt know that sorryy for pointless issue, also what are the chances they will look at the 2miners and see the command line arguments, should i be worried when i spread it massively?

**Comment by UnamSanctam** [comment]
>OHHHHH admin is needed i didnt know that sorryy for pointless issue, also what are the chances they will look at the 2miners and see the command line arguments, should i be worried when i spread it massively?

Well you don't really have any choice with lolminer so worrying about it wouldn't change much anyways. Some will notice though many probably won't.

**Comment by varysuzuki** [comment]
> > OHHHHH admin is needed i didnt know that sorryy for pointless issue, also what are the chances they will look at the 2miners and see the command line arguments, should i be worried when i spread it massively?
> 
> Well you don't really have any choice with lolminer so worrying about it wouldn't change much anyways. Some will notice though many probably won't.

what settings do u recommend to not notice any lag exactly like lolminer? or should i just say fuck it and go always 100%

**Comment by shainodev** [comment]
> > what settings do u recommend to not notice any lag exactly like lolminer? or should i just say fuck it and go always 100% (for ethminer)
> 
> Do you lag with 95%?

Oh, no i dont ill use that then

**Comment by shainodev** [comment]
> > what settings do u recommend to not notice any lag exactly like lolminer? or should i just say fuck it and go always 100% (for ethminer)
> 
> Do you lag with 95%?

I have a new problem now, when i try to Unamlib_encrypt the arguments part then the miner just  doesnt inject 
![image](https://user-images.githubusercontent.com/77803435/187032564-19a32cb7-b3c2-4225-a439-a70219c9107b.png)
if i dont do that then it does inject

**Comment by UnamSanctam** [comment]
Make sure that it has the identifier ID as well, it's like this: `conhost.exe IDENTIFIER ARGUMENTS`, the identifier is there to be able to find the miner.

**Comment by shainodev** [comment]
> Make sure that it has the identifier ID as well, it's like this: `conhost.exe IDENTIFIER ARGUMENTS`, the identifier is there to be able to find the miner. The identifier ID can be anything you want, in the miner it's random per build.

Yeah the ViTyQeCJa9SXs9 is my minerid i use to find it. i generated my own with the length of the minerFind and miner.nid what is 8 to 16 + 1. do i have to use --cinit-id=? cause i dont know what u mean could u give a better example of what i should do maybe remove the space? "ViTyQeCJa9SXs9" + Unamlib_Encrypt(argstr.ToString()) this? or what

**Comment by UnamSanctam** [comment]
No the space should be there, just do it exactly like the miner does it and it should work. Make sure you haven't changed the encryption keys.

**Comment by shainodev** [comment]
> No the space should be there, just do it exactly like the miner does it and it should work. Make sure you haven't changed the encryption keys.

Nope, they are still         
public string UNAMKEY = "UXUUXUUXUUCommandULineUUXUUXUUXU";
public string UNAMIV = "UUCommandULineUU";
i really dont know what to do it just doesnt want to work when i do Unamlib_Encrypt to the arguments
![image](https://user-images.githubusercontent.com/77803435/187033120-d9651f16-5fd2-47d6-b70c-2e5ffba03c5b.png)

**Comment by UnamSanctam** [comment]
Print the whole thing out to make sure that it's correct.

**Comment by shainodev** [comment]
> Print the whole thing out to make sure that it's correct.

![image](https://user-images.githubusercontent.com/77803435/187033264-93381a77-71ea-46de-8494-2248d2137f3a.png)
seems correct to me idk, oh should it be ViTyQeCJa9SXs9 "args"?

**Comment by UnamSanctam** [comment]
How does it look where you inject it?

**Comment by shainodev** [comment]
> 

wdym?
![image](https://user-images.githubusercontent.com/77803435/187033485-5b98a693-b970-4d4c-b10f-9c4ec51f2c04.png)
u mean this?

**Comment by shainodev** [comment]
ohhh i am not allowed to encrypt the minerid and the argument part together for some reason

**Comment by shainodev** [comment]
i have to encrypt it so it isnt together and then combine them when i _rInject_

**Comment by UnamSanctam** [comment]
Print it out in the miner after decrypting both together vs decrypting both separately and combining them together to see the difference.

**Comment by shainodev** [comment]
> Print it out in the miner after decrypting both together vs decrypting both separately and combining them together to see the difference.

theres no difference at all

**Comment by shainodev** [comment]
it just doesnt want to work when i combine them

**Comment by shainodev** [comment]
> Print it out in the miner after decrypting both together vs decrypting both separately and combining them together to see the difference.

hey, i looked at ur online download and read Don't use this function to download files directly from my GitHub. and then i thought what if i make it download directly from the lolminers release github instead of putting the miner manually on bitbucket? will they get banned? and is it a good idea, i thought it was not cause then the zip file is not encrypted but idk

**Comment by shainodev** [comment]
is it even better to make it download from the internet will it change anything?

**Comment by UnamSanctam** [comment]
>hey, i looked at ur online download and read Don't use this function to download files directly from my GitHub. and then i thought what if i make it download directly from the lolminers release github instead of putting the miner manually on bitbucket? will they get banned? and is it a good idea, i thought it was not cause then the zip file is not encrypted but idk

Yes it technically could get them banned, though so many download it that it probably won't happen. Shouldn't really matter if it's encrypted or not for Windows Defender (not sure if an AMSI bypass is needed for the .NET webclient), for other ones it might matter though.

>is it even better to make it download from the internet will it change anything?

Means that you'll have less embedded resources so yes it theoretically should stop any detections from that.

**Comment by varysuzuki** [comment]
> > hey, i looked at ur online download and read Don't use this function to download files directly from my GitHub. and then i thought what if i make it download directly from the lolminers release github instead of putting the miner manually on bitbucket? will they get banned? and is it a good idea, i thought it was not cause then the zip file is not encrypted but idk
> 
> Yes it technically could get them banned, though so many download it that it probably won't happen. Shouldn't really matter if it's encrypted or not for Windows Defender (not sure if an AMSI bypass is needed for the .NET webclient), for other ones it might matter though.

so its a better idea to encrypt the ethminer.zip and upload it to bitbucket?

also this question is kindof off topic but i was wondering what do you think about andrew tate?

and should i buy amadey or smoke loader to manually drop the miner or no?

**Comment by UnamSanctam** [comment]
>so its a better idea to encrypt the ethminer.zip and upload it to bitbucket?

Sure, you can encrypt lolminer or ethminer and upload it.

>also this question is kindof off topic but i was wondering what do you think about andrew tate?

Well I don't really like internet celebrities overall, just don't buy anything he's selling.

>and should i buy amadey or smoke loader to manually drop the miner or no?

Would that help with anything?

**Comment by varysuzuki** [comment]
> Would that help with anything?

that would help to 100% hold bots but idk really, i could instead make my own loader and then manually have a gpu vram checker, so if they have 3gb it mines the best algorithm with ethminer for the gpu, then if it has 4gb etchash and if it has 6 gb ethash (idk if eth mining will be ever possible after the PoW switch to PoS but people are talking about some eth1 fork or something)

**Comment by varysuzuki** [comment]
also you should add a feature inside the ethminer, {GPUNAME} so it puts the gpu name as the workername

**Comment by shainodev** [comment]
> Creating your own loader in C++ should be good enough if you're able to.

but idk if its a good idea, and also i always get mixed feelings i have a feeling the hashrates keep going down and stuff since at 5pm its like 40mh/s but at like 10pm its 180 mh/s

> Problem is that it sets the pool before it gets the GPU, and also it wouldn't show much since many pools have (had) worker name limits quite small so I have to truncate it.

you could just simply make it RTX_3080TI or GTX_1060 or RX_660 for amd, so if they have the same gpu it will be the same worker, so if i would have 500 miners then it wouldnt be 500 different worker names and stuff yk

**Comment by UnamSanctam** [comment]
>but idk if its a good idea, and also i always get mixed feelings i have a feeling the hashrates keep going down and stuff since at 5pm its like 40mh/s but at like 10pm its 180 mh/s

Well most people usually don't have their computers on all the time, so naturally hashrates will go up when they use it and down when they don't. Usually during the night (where they are from) it goes down.

>you could just simply make it RTX_3080TI or GTX_1060 or RX_660 for amd, so if they have the same gpu it will be the same worker, so if i would have 500 miners then it wouldnt be 500 different worker names and stuff yk

Isn't too easy to do that, the default names are usually some variation of `NVIDIA GeForce RTX 1060` so I'd have to try and hardcode some strings to remove/replace and stuff, and even then some would get past that and just get something like `NVIDIALapt` as the name.

**Comment by shainodev** [comment]
> Isn't too easy to do that, the default names are usually some variation of `NVIDIA GeForce RTX 1060` so I'd have to try and hardcode some strings to remove/replace and stuff, and even then some would get past that and just get something like `NVIDIALapt` as the name.

hmm, also should i still mine ethereum? and what is the best way to detect if the pc can mine eth and if not then etc

**Comment by shainodev** [comment]
and could i get banned for having too many worker names?

**Comment by UnamSanctam** [comment]
>hmm, also should i still mine ethereum? and what is the best way to detect if the pc can mine eth and if not then mine etc

No idea. Check the VRAM using either the registry or something like D3D9 (if using C++).

>and could i get banned for having too many worker names?

Technically yes, but I haven't gotten any reports about it on nanopool, 2miners and ethermine.

**Comment by shainodev** [comment]
> Technically yes, but I haven't gotten any reports about it on nanopool, 2miners and ethermine.

oh alright ill use it then. also what time consuming things do u have left with the c++ version of miner and the unamdownloader? since school is starting and ill have little to no time to really do anything and im kindof stressed

**Comment by UnamSanctam** [comment]
>oh alright ill use it then. also what time consuming things do u have left with the c++ version of miner and the unamdownloader? since school is starting and ill have little to no time to really do anything and im kindof stressed, and also what is the cinit-id used for?

Currently for the miner all that I think I have left is, 1) Finishing the file writing and file reading for the startup and watchdog 2) Finishing the watchdog 3) Implementing everything into the builder 4) Checking everything and modifying to further reduce detections.

The UnamDownloader is all done except the startup.

**Comment by varysuzuki** [comment]
> Currently for the miner all that I think I have left is, 1) Finishing the file writing and file reading for the startup and watchdog 2) Finishing the watchdog 3) Implementing everything into the builder 4) Checking everything and modifying to further reduce detections.
> 
> The UnamDownloader is all done except the startup.

great! also could you help me how to mine with proxies?

**Comment by shainodev** [comment]
> > great! also could you help me how to mine with proxies?
> 
> For xmrig there is an option for a SOCKS5 proxy, for ethminer there isn't any that I know of.

My c++ downloader for some reason needs VCRUNTIME140.dll when i open it on AppOnFly Free Trial and when i put VCRUNTIME140.dll there it says application cannot be started or something but when i use it on my pc then it works perfectly. why is this so?

**Comment by UnamSanctam** [comment]
You need to build it statically, you're building it dynamically with Visual C++ so to run it you would need to install Visual C++.

**Comment by varysuzuki** [comment]
> You need to build it statically, you're building it dynamically with Visual C++ so to run it you would need to install Visual C++.

how do i build it statically?

**Comment by UnamSanctam** [comment]
If you're using Visual Studio then you can right-click the project, press "Properties" and then press "C/C++". There you will find "Runtime Library", change that to "Multi-threaded (/MT)".
